### PR TITLE
Construction Window StationTab toolDown

### PR DESCRIPTION
**Pull request by duncanspumpkin**
The port placement is surprisingly complex as it works out the direction based on where there is water/industry.

### CODE REVIEW

**Review by AaronVanGeffen — COMMENTED** [review]
Having trouble building heliports. Airports work fine.

Edit: they seem to be building underground? The ghosts appear fine though.
Screenshot (1)

File: src/OpenLoco/Windows/Construction/StationTab.cpp [hunk 1]
```diff
@@ -167,15 +171,356 @@ namespace OpenLoco::Ui::Windows::Construction::Station
...
+                    continue;
+                }
+                const auto tile = TileManager::get(checkPos);
+                const auto* surface = tile.surface();
+                if (surface == nullptr)
+                {
+                    return std::nullopt;
+                }
+
+                const auto baseZ = surface->water() ? surface->water() * 4 : surface->baseZ();
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/Construction/StationTab.cpp:227]: Why is water taken into account here? We can't build an airport on water, right? (Does land surface not require the *4 here?)

Comment (duncanspumpkin) [reply src/OpenLoco/Windows/Construction/StationTab.cpp:227]: Water is a microZ value, baseZ is a smallZ. *4 not required for baseZ

**Review by AaronVanGeffen — APPROVED** [review]
Helipads work fine now

**Pull request merged**